substrate/primitives/staking/src/lib.rs

```diff
@@ -304,6 +310,33 @@ pub trait StakingInterface {
 	fn set_current_era(era: EraIndex);
 }
 
+/// Set of low level apis to manipulate staking ledger.
+///
+/// These apis bypass some or all safety checks and should only be used if you know what you are
+/// doing.
+pub trait StakingUnsafe: StakingInterface {
+	/// Migrate an existing staker to a virtual staker.
+	///
+	/// It would release all funds held by the implementation pallet.
+	fn migrate_to_virtual_staker(who: &Self::AccountId);
+
+	/// Book-keep a new bond for `who` without applying any locks (hence virtual).
+	///
+	/// It is important that who is a keyless account and therefore cannot interact with staking
+	/// pallet directly. Caller is responsible for ensuring the passed amount is locked and valid.
+	fn virtual_bond(
+		keyless_who: &Self::AccountId,
+		value: Self::Balance,
+		payee: &Self::AccountId,
```

Comment:
Seems like there are some constraints on `payee` as well that are not explained here.

Reply:
It is more of an implementation constraint and not a trait level constraint. 

Added doc to impl.